### PR TITLE
Yakov/fixes

### PR DESCRIPTION
**Pull request by yakov-g**

### CODE REVIEW

File: mlcomp/parallelm/mlpiper/component_scanner.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ def scan_dir(self, root_dir):
             comps[engine_type][comp_name]["comp_desc"] = comp_desc
             comps[engine_type][comp_name]["root"] = root
             comps[engine_type][comp_name]["files"] = self._include_files(root, comp_desc)
+            # always include current component json file regardless of its name
```
Comment (lior-amar-pm) [inline mlcomp/parallelm/mlpiper/component_scanner.py]: Capital letter at beginning of comment

File: mlcomp/parallelm/mlpiper/main.py [hunk 2]
```diff
@@ -127,6 +130,9 @@ def _add_deps_sub_parser(subparsers):
     group.add_argument('-f', '--file', type=argparse.FileType('r'),
                        help='A json file path, whose content is a pipeline. Or component JSON')
 
+    deps.add_argument('-r', '--comp-root', default=None, required=True,
+                                help='MLPiper components root dir. Recursively detecting components')
+
```
Comment (lior-amar-pm) [inline mlcomp/parallelm/mlpiper/main.py:135]: I wanted you to add a new option which will only show the dependancies not install them.,

File: mlcomp/parallelm/mlpiper/main.py [hunk 3]
```diff
@@ -127,6 +130,9 @@ def _add_deps_sub_parser(subparsers):
     group.add_argument('-f', '--file', type=argparse.FileType('r'),
                        help='A json file path, whose content is a pipeline. Or component JSON')
 
+    deps.add_argument('-r', '--comp-root', default=None, required=True,
```
Comment (lior-amar-pm) [inline mlcomp/parallelm/mlpiper/main.py:133]: Fix the comment "Return" to Print

**Review by lior-amar-pm — APPROVED** [review]

File: mlcomp/parallelm/mlpiper/component_scanner.py (hunk 1 above)
Comment (yakov-g) [reply mlcomp/parallelm/mlpiper/component_scanner.py]: done

File: mlcomp/parallelm/mlpiper/main.py (hunk 3 above)
Comment (yakov-g) [reply mlcomp/parallelm/mlpiper/main.py:133]: done

File: mlcomp/parallelm/mlpiper/main.py [hunk 4]
```diff
@@ -159,9 +172,27 @@ def main(bin_dir=None):
...
+        ml_piper = MLPiper(options) \
+            .comp_repo(options.comp_root) \
+            .deployment_dir(tmp_dir) \
+            .bin_dir(bin_dir) \
+            .pipeline(options.pipeline if options.pipeline else options.file) \
+            .use_color(not options.no_color) \
+            .force(True)
+
+        ml_piper.deps(options.lang)
+        if os.path.exists(tmp_dir) and os.path.isdir(tmp_dir):
```
Comment (zohar-mizrahi) [inline mlcomp/parallelm/mlpiper/main.py]: 1. `os.path.exists` is redundant
2. You may use `try` & `finally` to ensure tmp dir removal

File: mlcomp/parallelm/mlpiper/main.py [hunk 5]
```diff
@@ -106,18 +109,24 @@ def _add_deploy_sub_parser(subparsers, sub_parser_name, sub_parser_help):
     parser_prepare.add_argument('--force', action='store_true',
                                 help='Overwrite any previous generated files/directories (.e.g deployed dir)')
 
+    parser_prepare.add_argument('--test-mode', default=False, required=False, action="store_true",
```
Comment (zohar-mizrahi) [inline mlcomp/parallelm/mlpiper/main.py]: What is the purpose of the "test-mode"?

File: mlcomp/parallelm/pipeline/executor.py [hunk 6]
```diff
@@ -103,13 +103,22 @@ def all_component_dependencies(self, lang):
...
+                                                         comp_root_path=self._comp_root_path).load(extended=True)
         for comp_desc in comps_desc_list:
             if comp_desc[json_fields.PIPELINE_LANGUAGE_FIELD] == lang:
                 deps = comp_desc.get(json_fields.COMPONENT_DESC_PYTHON_DEPS, None)
                 if deps:
                     accumulated_py_deps.update(deps)
 
+                # for Python fetch deps from requirements.txt file
+                if lang == ComponentLanguage.PYTHON:
+                    req_file = os.path.join(comp_desc[json_fields.COMPONENT_DESC_ROOT_PATH_FIELD], "requirements.txt")
```
Comment (zohar-mizrahi) [inline mlcomp/parallelm/pipeline/executor.py]: Don't we need to make sure to include it with the component's files when we scan the files for a given component? Is it required to be placed it in the `includeGlobPattern`?

**Review by zohar-mizrahi — APPROVED** [review]

File: mlcomp/parallelm/pipeline/executor.py (hunk 6 above)
Comment (yakov-g) [reply mlcomp/parallelm/pipeline/executor.py]: yes, for now I require to place it into include patterns, but I don't mind to always include it.

**Pull request merged**